pandas/tests/indexing/test_indexing.py

```diff
@@ -83,16 +80,16 @@ def test_getitem_ndarray_3d(self, index, obj, idxr, idxr_id):
                 "Cannot index with multidimensional key",
                 r"Wrong number of dimensions. values.ndim != ndim \[3 != 1\]",
                 "Index data must be 1-dimensional",
+                "positional indexers are out-of-bounds",
+                "Indexing a MultiIndex with a multidimensional key is not implemented",
             ]
         )
 
-        with pytest.raises(ValueError, match=msg):
+        potential_errors = (IndexError, ValueError, NotImplementedError)
+        with pytest.raises(potential_errors, match=msg):
```

Comment:
I tried splitting the errors and messages to attribute them to the individual cases. That proved to be _very_ tidious and confusing though. I also don't think it provides a lot of value. After all, the purpose of this test is to verify that this is raising